src/Properties/properties.cpp

```diff
@@ -38,4 +39,101 @@ Vector<double, 3> calculateElectronicDipoleMoment(const VectorSQOneElectronOpera
+            throw std::runtime_error("properties::calculateDysonOrbital(WaveFunction, WaveFunction): wavefunction2 is not expressed in a product Fock space with one fewer electron than wavefunction1");
+        }
+    } 
+
+    // Initialize environment variables
+
+    // The 'passive' spin Fock spaces are the Fock spaces that are equal for both wave functions
+    // The 'target' spin Fock spaces have an electron difference of one
+    // We initialize the variables for the case in which they differ in one beta electron, if this isn't the case, we will update it later    auto passive_fock_space1 = fock_space1.get_fock_space_alpha();
+    auto passive_fock_space1 = fock_space1.get_fock_space_alpha();
+    auto passive_fock_space2 = fock_space2.get_fock_space_alpha();
+    auto target_fock_space1 = fock_space1.get_fock_space_beta();
+    auto target_fock_space2 = fock_space2.get_fock_space_beta();
+
+    // Mod variables relate to the modification of the address jump in coefficient index according to the ordering of the spin ONVs 
+    size_t passive_mod1 = target_fock_space1.get_dimension();
+    size_t passive_mod2 = target_fock_space2.get_dimension();
+    size_t target_mod = 1;
+
+    // If instead the Fock spaces differ by one alpha electron we re-assign the variables to match the algorithm
+    if ((fock_space1.get_N_alpha() - fock_space2.get_N_alpha() == 1) && (fock_space1.get_N_beta() - fock_space2.get_N_beta() == 0)) {
+        passive_fock_space1 = target_fock_space1;
+        passive_fock_space2 = target_fock_space2;
+        target_fock_space1 = fock_space1.get_fock_space_alpha();
+        target_fock_space2 = fock_space2.get_fock_space_alpha();
+        
+        passive_mod1 = 1;
+        passive_mod2 = 1;
+        target_mod = passive_fock_space1.get_dimension();
+    }
+
+    const auto& ci_coeffs1 = wavefunction1.get_coefficients();
+    const auto& ci_coeffs2 = wavefunction2.get_coefficients();
+
+    VectorX<double> dyson_coeffs = VectorX<double>::Zero(fock_space1.get_K());
+
+    // The actual algorithm to determine the Dyson amplitudes
+    // Since we want to calculate the overlap between two wave functions, the ONVs should have an equal number of electrons
+    // We therefore iterate over the ONVs of the 'target' Fock space, which all have an electron more, and annihilate in one of the orbitals (let the index of that orbital be called 'p')
+    // By calculating the overlap in the (N-1)-Fock space, we can calculate the contributions to the  'p'-th coefficient (i.e. the Dyson amplitude) of the Dyson orbital
+    ONV onv = target_fock_space1.makeONV(0);
+
+    for (size_t It = 0; It < target_fock_space1.get_dimension(); It++) {  // It loops over addresses of the target Fock space
+        int sign = -1;  // total phase factor of all the annihilations that have occurred
+        for (size_t e = 0; e < target_fock_space1.get_N(); e++) {  // loop over electrons in the ONV
+            
+            sign *= -1; 
```

Comment:
Could you move lines 112 and 113 below the comment that starts on 115? In this way, the code is more structured.